index.ts

```diff
@@ -0,0 +1,552 @@
+interface MoveFlags {
+  authentic?: boolean; // Ignores a target's substitute.
+  bite?: boolean; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Strong Jaw.
+  bullet?: boolean; // Has no effect on Pokemon with the Ability Bulletproof.
+  charge?: boolean; // The user is unable to make a move between turns.
+  contact?: boolean; // Makes contact.
+  dance?: boolean; // When used by a Pokemon, other Pokemon with the Ability Dancer can attempt to execute the same move.
+  defrost?: boolean; // Thaws the user if executed successfully while the user is frozen.
+  distance?: boolean; // Can target a Pokemon positioned anywhere in a Triple Battle.
+  gravity?: boolean; // Prevented from being executed or selected during Gravity's effect.
+  heal?: boolean; // Prevented from being executed or selected during Heal Block's effect.
+  mirror?: boolean; // Can be copied by Mirror Move.
+  mystery?: boolean; // Unknown effect.
+  nonsky?: boolean; // Prevented from being executed or selected in a Sky Battle.
+  powder?: boolean; // Has no effect on Pokemon which are Grass-type, have the Ability Overcoat, or hold Safety Goggles.
+  protect?: boolean; // Blocked by Detect, Protect, Spiky Shield, and if not a Status move, King's Shield.
+  pulse?: boolean; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+  punch?: boolean; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+  recharge?: boolean; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+  reflectable?: boolean; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+  snatch?: boolean; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+  sound?: boolean; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+  // single-target
+  'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+  // single-target, automatic
+  'self'|'randomNormal'|
+  // spread
+  'allAdjacent'|'allAdjacentFoes'|
+  // side and field
+  'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+  volatileStatus?: ID;
+  boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+  chance?: number;
+  boosts?: Partial<BoostsTable>;
+  self?: SelfEffect;
+  status?: StatusName;
+  volatileStatus?: ID;
+}
+
+type ContestType = 'Cool' | 'Beautiful' | 'Cute' | 'Clever' | 'Tough';
+
+interface CommonMove<TypeT, ItemT, MoveT> {
+  accuracy: number | true;
+  basePower: number;
+  flags: MoveFlags;
+  pp: number;
+  target: MoveTarget;
+  type: TypeT;
+
+  alwaysCrit?: boolean;
+  alwaysHit?: boolean;
+  basePowerModifier?: number;
+  boosts?: Partial<BoostsTable> | false
+  bypassesProtect?: boolean;
+  category: MoveCategory;
+  contestType?: ContestType;
+  critModifier?: number;
```

Comment:
like `basePowerModifier`, this is used in calculations in a few places, but doesn't actually seem to be on any data files. Am I missing something here?

Reply:
They're both ActiveMove properties, I'd assume.

Reply:
Not even though - they appear to just literally never be used. eg:

```
kjs@scheibo(master)$ gg critModifier
data/mods/gen4/scripts.js:                        baseDamage = this.modify(baseDamage, move.critModifier || 2);
data/mods/ssb/scripts.js:                 baseDamage = tr(baseDamage * (move.critModifier || (this.gen >= 6 ? 1.5 : 2)));
sim/battle.ts:                    baseDamage = tr(baseDamage * (move.critModifier || (this.gen >= 6 ? 1.5 : 2)));
sim/dex-data.ts:  readonly critModifier?: number;
sim/globals.ts:   critModifier?: number
```

Is what you get for grepping from the root, `basePowerModifier` is similar. Were these like used for a past OM or something?

Reply:
Most likely yes.

Reply:
Although also likely is that Sniper used to use it but was refactored to do something else.

Also why `gg`? Use `rg` like a normal person. :p

Reply:
`git grep` has a different output format which is clickable in VS Code. fsr I cant seem to configure `rg` to do it. Also, years of typing `gg` have instilled muscle memory that is hard to erase, even if I could figure out how to make `rg` work for me I'd still alias it to `gg`.